addons/website_sale/models/sale_order.py

```diff
@@ -318,11 +318,15 @@ def _cart_add(self, product_id: int, quantity: float = 1.0, *, uom_id: int | Non
             uom_id = self.env['product.product'].browse(product_id).uom_id.id  # type: ignore
         if existing_sol := self._cart_find_product_line(product_id, uom_id=uom_id, **kwargs)[:1]:
             # If a matching line is found, update the existing line instead.
-            return self._cart_update_line_quantity(
+            values = self._cart_update_line_quantity(
                 line_id=existing_sol.id,  # type: ignore
                 quantity=existing_sol.product_uom_qty + quantity,
                 **kwargs,
             )
+            return {
+                'added_qty': values['quantity'] - existing_sol.product_uom_qty,  # new_qty - old_qty
```

Comment:
The SOL has been updated, so `added_qty` is 0. I think adapting `_cart_update_line_quantity` would be easier.